### DISCUSSION TITLE
Migrate LFS repo from Jfrog artifactory repo to Bitbucket cloud

### DISCUSSION THREAD

**Discussion by rgaduput** [git-lfs/git-lfs#5567] [Q&A]
Hi,
Currently we have repositories in Bitbucket cloud which are enabled LFS, LFS content is being pushed into our private jfrog artifactory repository. This is done by adding the `.lfsconfig` on git repo level which contains `lfs.url` of remote repository.
Now we would like to move the LFS content also to Bitbucket cloud.

Is it possible to perform the migration to cloud ?

Another thing which is bit unclear to me is that Bitbucket cloud does not require any `lfs.url` we just need to add `git-lfs add` or `.gitattributes`. So even if managed to to pull/push LFS content from cloud by removing the file `.lfsconfig` for current branches but what happens to the new checkouts done on old commits/tags where this file will exists and points to the wrong LFS location ?

Appreciate any suggestion to overcome this.

Thanks.

**Comment by bk2204** [comment]
It's possible to push data to another repository simply by specifying the remote explicitly.  For example, you can do this to migrate the data to a new remote (assuming `bitbucket` is your new remote):

```
$ git fetch
$ git lfs fetch --all
$ git lfs push --all bitbucket
```

If you're converting to a new remote which is on the same server as the Git remote (which it sounds like you are), then you don't need to specify `lfs.url` in `.lfsconfig`.  You'd want to remove that entry (or the file altogether) to avoid pushing problems.  You can also rewrite URLs for `lfs.url` by setting `lfs.transfer.enablehrefrewrite` and use the `url.*.insteadOf` functionality (which is part of Git and documented in `git-config(5)`) to automatically rewrite the older URLs to the newer ones.  That will need to be done on each individual machine, though.
